Re-scaffold k0sproject/k0s (support cosign)

takumin opened the pull request:
[k0sproject/k0s](https://github.com/k0sproject/k0s): k0s - The Zero Friction Kubernetes

```console
$ aqua g -i k0sproject/k0s
```

## Check List

<!-- Please check the list. Please don't remove the check list. -->

- [x] Read [CONTRIBUTING.md](https://aquaproj.github.io/docs/products/aqua-registry/contributing)
  - [x] Read [OSS Contribution Guide](https://github.com/suzuki-shunsuke/oss-contribution-guide/blob/main/README.md)
- [x] [All commits are signed](https://github.com/suzuki-shunsuke/oss-contribution-guide/blob/main/docs/commit-signing.md)
  - This repository enables `Require signed commits`, so all commits must be signed
- [x] [Avoid force push](https://github.com/suzuki-shunsuke/oss-contribution-guide?tab=readme-ov-file#dont-do-force-pushes-after-opening-pull-requests)
- [x] Do only one thing in one Pull Request
- [x] [Execute cmdx s to scaffold code](https://aquaproj.github.io/docs/products/aqua-registry/contributing/#use-cmdx-s-definitely)
- [x] Install and execute the package and confirm if the package works well

## How to confirm if this package works well

Reviewers aren't necessarily familiar with this package, so please describe how to confirm if this package works well.
Please confirm if this package works well yourself as much as possible.

Command and output

```console
$ cmdx con linux amd64
+ bash scripts/connect.sh
[INFO] Connecting to the container aqua-registry (linux/amd64)
root@bec87e916077:/workspace# k0s version
v1.32.3+k0s.0
```

If files such as configuration file are needed, please share them.

Reference

- https://docs.k0sproject.io/stable/
- https://docs.k0sproject.io/stable/verifying-signs/
- https://docs.k0sproject.io/stable/install/

Note

Please note the registry comments below:

> `--insecure-ignore-tlog` is required because it is signed with `--tlog-upload=false` option

- sign for linux/amd64
  - https://github.com/k0sproject/k0s/blob/v1.32.3%2Bk0s.0/.github/workflows/release.yml#L106
- sign for linux/arm64
  - https://github.com/k0sproject/k0s/blob/v1.32.3%2Bk0s.0/.github/workflows/release.yml#L266
- sign for windows/amd64
  - https://github.com/k0sproject/k0s/blob/v1.32.3%2Bk0s.0/.github/workflows/release.yml#L198

takumin committed:
feat(k0sproject/k0s): scaffold k0sproject/k0s

takumin committed:
fix(k0sproject/k0s): pass cmdx t

suzuki-shunsuke commented on pkgs/k0sproject/k0s/registry.yaml:
```diff
@@ -3,13 +3,41 @@ packages:
+            type: github_release
+            asset: "cosign.pub"
+          opts:
+            # `--insecure-ignore-tlog` is required because it is signed with `--tlog-upload=false` option
+            # see also: https://github.com/k0sproject/k0s/blob/v1.32.3%2Bk0s.0/.github/workflows/release.yml#L106
+            - --insecure-ignore-tlog
```
- https://github.com/k0sproject/k0s/commit/a01579db2dc7916194e6fd95949b6b4fbbfa2252
- https://github.com/k0sproject/k0s/pull/5724

Seems like this will be back ported to 1.29 or later.

takumin replied:
Supports all currently released versions.

https://github.com/aquaproj/aqua-registry/pull/35740/commits/54bee550cb44d030489f477fd75220316af9a1bc

takumin committed:
fix(k0sproject/k0s): support cosign transparency log

Transparency logs are now uploaded upstream.

see also:
https://github.com/k0sproject/k0s/pull/5724
https://github.com/k0sproject/k0s/commit/a01579db2dc7916194e6fd95949b6b4fbbfa2252

suzuki-shunsuke commented:
🎉 https://github.com/aquaproj/aqua-registry/releases/tag/v4.357.0
